packages/clarity-js/src/data/metadata.ts

```diff
@@ -20,6 +21,9 @@ export function start(): void {
     pageNum: s[1]
   }
 
+  // Check if the session should start off in full mode based on the signal from session storage
+  config.lean = config.track && sessionStorage && sessionStorage.getItem(Constant.UPGRADE_KEY) ? false : config.lean;
```

Comment:
Why is the first case of the : false? If the condition is met should the lean mode not be turned to on?
Also is it safe to override the config as you do here?

Reply:
Because if the session was previously upgraded we want the lean mode to be off (false) for this session. If the session was not already upgraded then we honor whatever current config says. Yes, safe to do so.